package/kernel/linux/modules/lib.mk

```diff
@@ -98,6 +98,132 @@ endef
 $(eval $(call KernelPackage,lib-crc32c))
 
 
+define KernelPackage/lib-crypto-blake2s
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=BLAKE2s hash function library
+  KCONFIG:=CONFIG_CRYPTO_LIB_BLAKE2S
+  FILES:= \
+	$(LINUX_DIR)/lib/crypto/libblake2s.ko \
+	$(LINUX_DIR)/lib/crypto/libblake2s-generic.ko
+endef
+
+define KernelPackage/lib-crypto-blake2s/x86/64
+  KCONFIG+=CONFIG_CRYPTO_BLAKE2S_X86
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/blake2s-x86_64.ko
+endef
+
+$(eval $(call KernelPackage,lib-crypto-blake2s))
+
+
+define KernelPackage/lib-crypto-chacha20
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=ChaCha library interface
+  KCONFIG:=CONFIG_CRYPTO_LIB_CHACHA
+  FILES:=$(LINUX_DIR)/lib/crypto/libchacha.ko
+endef
+
+define KernelPackage/lib-crypto-chacha20/x86/64
+  KCONFIG+=CONFIG_CRYPTO_CHACHA20_X86_64
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/chacha-x86_64.ko
+endef
+
+define KernelPackage/lib-crypto-chacha20/arm-neon
```

Comment:
Can you just call this "arm" instead of "arm-neon"? The accelerated code has paths for both neon and !neon systems.

Reply:
okay, even though the CONFIG flag states NEON?

Reply:
tl;dr of below is the answer to your question is, "yes."

Here's the config flag definition:

```
config CRYPTO_CHACHA20_NEON
        tristate "NEON and scalar accelerated ChaCha stream cipher algorithms"
        select CRYPTO_BLKCIPHER
        select CRYPTO_ARCH_HAVE_LIB_CHACHA
```

This one does *not* have `depends on KERNEL_MODE_NEON` like some of the others do. And inside the actual glue code, you can see it picks two separate paths, depending on the platform:

```c
        if (!IS_ENABLED(CONFIG_KERNEL_MODE_NEON) || !neon_usable() ||
            bytes <= CHACHA_BLOCK_SIZE) {
                chacha_doarm(dst, src, bytes, state, nrounds);
                state[12] += DIV_ROUND_UP(bytes, CHACHA_BLOCK_SIZE);
                return;
        }

        do {
                unsigned int todo = min_t(unsigned int, bytes, SZ_4K);

                kernel_neon_begin();
                chacha_doneon(state, dst, src, todo, nrounds);
                kernel_neon_end();

                bytes -= todo;
                src += todo;
                dst += todo;
        } while (bytes);
```

Reply:
fixed